src/dbnode/persist/fs/files.go

```diff
@@ -1012,13 +1012,28 @@ func IndexSnapshotFiles(filePathPrefix string, namespace ident.ID) (FileSetFiles
 
 // FileSetAt returns a FileSetFile for the given namespace/shard/blockStart/volume combination if it exists.
 func FileSetAt(filePathPrefix string, namespace ident.ID, shard uint32, blockStart time.Time, volume int) (FileSetFile, bool, error) {
+	pattern := filesetFileForTimeAndIndex(blockStart, volume, anyLowerCaseCharsPattern)
```

Comment:
Fixing this method to support finding both legacy and non-legacy filesets (which is important as this method is called by `DeleteFileSetAt` which the merger uses for cleanup)

Reply:
Would it be worth not computing this in the case that it is legacy? I.e. just have `var pattern string` defined here, then after you do the volume check, `if len(pattern) == 0 { pattern = filesetFileForTimeAndIndex(blockStart, volume, anyLowerCaseCharsPattern) }`? Then probably don't need the fix to the base method since legacy patterns will already be parsed

Reply:
nit: should we call this `fileSetForTimeAndVolumeIndex` or `fileSetForTImeAndVolume`? `Index` here seems kind ambiguous.

Reply:
Yeah, we can defer setting pattern a bit. Will fix up